### PR TITLE
Fb1378

### PR DESCRIPTION
**Pull request by spireken**

### CODE REVIEW

File: Applications/spire/source/charting/charting_window.cpp [hunk 1]
```diff
@@ -226,3 +244,34 @@ void ChartingWindow::on_period_line_edit_changed() {
     m_period_dropdown->set_items({tr("seconds"), tr("minutes"), tr("hours")});
   }
 }
+
+void ChartingWindow::on_security_change(const Security& security) {
+  setWindowTitle(CustomVariantItemDelegate().displayText(
```
Comment (kranar) [inline Applications/spire/source/charting/charting_window.cpp]: ChartingWindow doesn't create the model, your test application has to. Propagate this event upward.

Comment (spireken) [reply Applications/spire/source/charting/charting_window.cpp]: Should I create the tester then, or put it in main?

**Comment by kranar** [comment]
Main is fine.

File: Applications/spire/source/charting/charting_window.cpp [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #include "spire/charting/charting_window.hpp"
 #include <climits>
+#include <random>
```
Comment (kranar) [inline Applications/spire/source/charting/charting_window.cpp]: Remove the unused header.

File: Applications/spire/source/charting/charting_window.cpp [hunk 3]
```diff
@@ -8,6 +10,7 @@
 #include <QVBoxLayout>
 #include "spire/charting/chart_view.hpp"
 #include "spire/charting/empty_chart_model.hpp"
+#include "spire/charting/local_chart_model.hpp"
```
Comment (kranar) [inline Applications/spire/source/charting/charting_window.cpp]: Remove this include.

File: Applications/spire/source/charting/charting_window.cpp [hunk 4]
```diff
@@ -8,6 +10,7 @@
 #include <QVBoxLayout>
 #include "spire/charting/chart_view.hpp"
 #include "spire/charting/empty_chart_model.hpp"
```
Comment (kranar) [inline Applications/spire/source/charting/charting_window.cpp]: You might need to remove this header too.

File: Applications/spire/source/charting/charting_window.cpp [hunk 5]
```diff
@@ -32,6 +36,7 @@ ChartingWindow::ChartingWindow(Ref<SecurityInputModel> input_model,
       m_model(std::make_shared<EmptyChartModel>(ChartValue::Type::TIMESTAMP,
```
Comment (kranar) [inline Applications/spire/source/charting/charting_window.cpp]: You likely don't need to initialize this model anymore.

File: Applications/spire/source/charting/charting_window.cpp [hunk 6]
```diff
@@ -124,27 +132,28 @@ ChartingWindow::ChartingWindow(Ref<SecurityInputModel> input_model,
...
   m_security_widget->setFocus();
-  m_chart = new ChartView(*m_model, this);
-  m_security_widget->set_widget(m_chart);
-  m_chart->installEventFilter(this);
 }
 
 void ChartingWindow::set_model(std::shared_ptr<ChartModel> model) {
   m_model = model;
-  delete m_chart;
+  if(m_chart != nullptr) {
```
Comment (kranar) [inline Applications/spire/source/charting/charting_window.cpp]: Don't check for ```nullptr``` when calling ```delete```.

**Pull request merged**